frontend/model/contracts/group.js

```diff
@@ -100,20 +100,22 @@ function clearOldPayments ({ contractID, state, getters }) {
     delete state.paymentsByPeriod[period]
   }
 
-  sbp('gi.contracts/group/pushSideEffect', contractID,
-    ['gi.contracts/group/archivePayments', contractID, archivingPayments]
-  )
+  if (direction === 'incoming') {
+    sbp('gi.contracts/group/pushSideEffect', contractID,
+      ['gi.contracts/group/archivePayments', contractID, archivingPayments]
+    )
+  }
```

Comment:
Instead of making all of these changes, it would be simpler to clear the sideEffect stack for outgoing processings

Reply:
Yeah, I think so too. This is a temporary solution and I wanted @corrideat to solve this after he come back.
But, what I really want is to remove calling `processMessage` before publishEvent.

Reply:
@Silver-IT Actually.... we cannot clear the sideEffect stack... because things are asynchronous and the process function is acting on the `publish:<contractID>` queue, and incoming messages are processed on a different queue `<contractID>`, clearing the sideEffect stack in publishEvent could break the app (if a contract is being processed at the same time a different event is being published).

So this comment that @corrideat made is wrong:

```js
          // Process message to ensure that it is valid. Should this thow,
          // we propagate the error.
          // Because of this, 'chelonia/private/in/processMessage' SHOULD NOT
          // change the global Chelonia state and it MUST NOT call any
          // side-effects or change the global state in a way that affects
          // the meaning of any future messages or successive invocations.
          await sbp('chelonia/private/in/processMessage', entry, cloneDeep(state || {}))
```

In fact, because contracts could push to the sideEffect stack, they do in fact "change the global Chelonia state".

I agree that it would be better if events weren't processed when they were outgoing. However, @corrideat seems to insist it's necessary. When he is back I'd like to go over this with him one more time.

In the meantime it seems that your changes are unfortunately necessary.

Reply:
OTOH, maybe I'm mistaken, I missed this part above:

```js
        const newEntry = await sbp('chelonia/private/queueEvent', contractID, async () => {
```

This makes it so that the `processMessage` is called on the same queue as incoming events that are being handled. This means that maybe it is possible to clear the `this.sideEffectStacks[contractID]` stack...

Reply:
I've submitted a PR to clear the sideEffect stack before messages are processed, so sideEffects that are pushed by publishing outgoing messages shouldn't affect the processing of other incoming messages: https://github.com/okTurtles/group-income/pull/2205

Reply:
Thanks!!! :+1: